src/reactive_system_urc/src/ReactiveSystemPath.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * Created by William Gu on Nov 3 2018
+ * Implementation for Reactive System Pathfinder
+ */
+
+#include <ReactiveSystemPath.h>
+
+nav_msgs::Path
+ReactiveSystemPath::getPath(mapping_msgs_urc::RiskAreaArray risk_areas,
+                            sb_geom_msgs::Point2D goal_pos,
+                            float traj_time_inc,
+                            int traj_num_incs,
+                            float linear_vel,
+                            float max_angular_vel,
+                            int num_angular_vel,
+                            float risk_dist_tol_sq) {
+    nav_msgs::Path path;
+
+    // Find lowest trajectory score given the range of angular velocities we
+    // search
+    float min_traj_score = std::numeric_limits<float>::max();
+    std::vector<sb_geom_msgs::Point2D> best_traj;
+
+    for (int i = -num_angular_vel; i <= num_angular_vel; i++) {
+        float angular_vel = i * (max_angular_vel / num_angular_vel);
+        std::vector<sb_geom_msgs::Point2D> trajectory =
+        getArcTrajectory(linear_vel, angular_vel, traj_time_inc, traj_num_incs);
+        float traj_score =
+        getTrajectoryScore(trajectory, goal_pos, risk_areas, risk_dist_tol_sq);
+
+        if (traj_score < min_traj_score) {
+            min_traj_score = traj_score;
+            best_traj      = trajectory;
+        }
+    }
```

Comment:
Could we add a few more comments to this `for` loop to make things a bit more clear? If you're not sure what to add, no worries, I can probably make some suggestions.